src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -268,16 +279,15 @@ static File makeChunkFile(File baseDir, int i)
   public static class Outer implements SmooshedWriter
   {
     private final int fileNum;
-    private final OutputStream out;
     private final int maxLength;
+    private final GatheringByteChannel channel;
 
-    private boolean open = true;
     private int currOffset = 0;
 
-    Outer(int fileNum, OutputStream out, int maxLength)
+    Outer(int fileNum, FileChannel channel, int maxLength)
```

Comment:
Don't see the point of having two separate SmoothedWriter impls: Outer and here: https://github.com/metamx/java-util/pull/46/files#diff-99d3b3a2a7085d596b82e8831f793554R163. Why this is not just a single class, to remove boilerplate?